### PR TITLE
getBuyIntentToReplaceSkus() support

### PR DESCRIPTION
**Pull request by mzestars**
Update IInAppBillingService.aidl to latest from Google
Added code for to check if change subscription is supported
Added code for changing subscription

### CODE REVIEW

File: library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java [hunk 1]
```diff
@@ -187,14 +188,47 @@ public boolean loadOwnedPurchasesFromGoogle() {
...
+	 *
+	 * @param oldProductId passing null or empty string will act the same as {@link #subscribe(Activity, String)}
+	 */
+	public boolean updateSubscription(Activity activity, String oldProductId, String productId) {
+		List<String> oldProductIds = null;
+		if (!TextUtils.isEmpty(oldProductId)) {
+			oldProductIds = new ArrayList<>(1);
+			oldProductIds.add(oldProductId);
+		}
+		return updateSubscription(activity, oldProductIds, productId);
```
Comment (serggl) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: @mzestars maybe we should call here `isSubscriptionUpdateSupported()` method first?

Comment (mzestars) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: @serggl Sure.

Comment (serggl) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: @mzestars any update on this?

Comment (mzestars) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: @serggl This is my first time contributing to open source project. So, I make the changes and make another pull request?

Comment (serggl) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: Just make a change to your branch and it will be automaticslly picked up by
the current PR

сб, 16 янв. 2016 г., 18:28, Mohd Zulfadhli Bin Ab Rashid <
notifications@github.com>:

> In library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java
> <https://github.com/anjlab/android-inapp-billing-v3/pull/142#discussion_r49932981>
> :
>
> > +		}
> > +		return false;
> > +	}
> > +
> > +	/**
> > +	 * Change subscription i.e. upgrade or downgrade
> > +	 *
> > +	 * @param oldProductId passing null or empty string will act the same as {@link #subscribe(Activity, String)}
> > +	 */
> > +	public boolean updateSubscription(Activity activity, String oldProductId, String productId) {
> > +		List<String> oldProductIds = null;
> > +		if (!TextUtils.isEmpty(oldProductId)) {
> > +			oldProductIds = new ArrayList<>(1);
> > +			oldProductIds.add(oldProductId);
> > +		}
> > +		return updateSubscription(activity, oldProductIds, productId);
>
> @serggl <https://github.com/serggl> This is my first time contributing to
> open source project. So, I make the changes and make another pull request?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anjlab/android-inapp-billing-v3/pull/142/files#r49932981>
> .
>
-- 
Sergey, AnjLab

Comment (mzestars) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: @serggl Done.

**Comment by serggl** [comment]
thanks, merged.
